### ISSUE TITLE
Using S3 Containers 

### ISSUE THREAD

**Issue by joepages** [yiisoft/yii2-imagine#72]
I have a while working with  this  extension and never had the need of uploading the image directly to an S3 container. 

I’m wondering if there’s a way of modifying the `save()` method to upload the file directly to S3 instead of saving locally and then upload to S3. 

Specifically im doing `Imagine::thumbnail($image, $width, $height)->save()` 

any ideas?

**Comment by joepages** [comment]
Question was closed but no insight provided. Id really appreciate any help you guys may provide.

**Comment by schmunk42** [comment]
I think it's not in the scope of this library, you'd rather need something like https://github.com/creocoder/yii2-flysystem

**Comment by joepages** [comment]
I already have the integration to upload the files to S3. What I’m looking is a way to rewrite the `save()` method in this library to instead of saving the thumbs locally to be saved in s3 with my component.

**Comment by joepages** [comment]
Thanks for the information. 

After research I've found the Imagine actually returns the thumbnail when requested. So I've decided to go this route and test response times: 

`
$thumb = Imagine::thumbnail("{$this->getS3Url($this->url)}", $width, $height);

$this->getS3Component()->commands()->upload(
                "$s3Path/$filename-{$width}x{$height}.$extension",
                $thumb
            )
            ->withContentType($mainFile->type)
            ->execute();
`

This actually creates the thumbnail and then I'm able to put that Object directly into S3 bucket. 

I agree with you that for thumbs creation it should be faster fetch the image from the server itself but when having multiple servers and load balancers you actually can't tell the server where the image was stored. 

For now, and until my dev-ops can take a look at the server setup I think we will evaluate the performance of this approach.
